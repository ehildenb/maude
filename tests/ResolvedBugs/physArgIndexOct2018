#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/physArgIndexOct2018.maude -no-banner -no-advise \
  > physArgIndexOct2018.out 2>&1

diff $srcdir/physArgIndexOct2018.expected physArgIndexOct2018.out > /dev/null 2>&1
