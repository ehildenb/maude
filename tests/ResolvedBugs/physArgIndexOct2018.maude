set show timing off .
set show advisories off .

***
***	Check that we are not confusing nominal and physical
***	argument indices for AC(U) terms with arguments having
***	multiplicity > 1.
***

mod VENDING-MACHINE is
  sorts Coin Item Place Marking .
  subsorts Coin Item < Place < Marking .
  op __ : Marking Marking -> Marking [ctor assoc comm id: null] .
  op null : -> Marking [ctor] .
  op $ : -> Coin [ctor format (r! o)] .
  op q : -> Coin [ctor format (r! o)] .
  op a : -> Item [ctor format (b! o)] .
  op c : -> Item [ctor format (b! o)] .

  rl [buy-c] : $ => c .
  rl [buy-a] : $ => a  q .
  rl [change]: q  q  q  q => $ .
endm

*** bad extra results
srewrite in VENDING-MACHINE : $ $ $ using all .

*** runaway execution
search $ $ $ =>! a a M:Marking .

mod FOO is
  sort Foo .
  op __ : Foo Foo -> Foo [assoc comm] .
  ops a b c : -> Foo .

  rl b => c .
endm

*** runaway execution
rew a a b .
