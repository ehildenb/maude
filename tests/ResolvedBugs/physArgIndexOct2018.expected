==========================================
srewrite in VENDING-MACHINE : $ $ $ using all .

Solution 1
rewrites: 1
result Marking: $ $ c

Solution 2
rewrites: 2
result Marking: $ $ q a

No more solutions.
rewrites: 2
==========================================
search in VENDING-MACHINE : $ $ $ =>! a a M:Marking .

Solution 1 (state 8)
states: 10  rewrites: 12
M:Marking --> q q c

Solution 2 (state 9)
states: 10  rewrites: 12
M:Marking --> q q q a

No more solutions.
states: 10  rewrites: 12
==========================================
rewrite in FOO : a a b .
rewrites: 1
result Foo: a a c
Bye.
